kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/StartProcessServiceIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+*/
+
+package org.kie.server.integrationtests.jbpm;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kie.api.KieServices;
+import org.kie.internal.KieInternalServices;
+import org.kie.internal.process.CorrelationKey;
+import org.kie.internal.process.CorrelationKeyFactory;
+import org.kie.internal.runtime.conf.RuntimeStrategy;
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.api.model.KieServerConfigItem;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.api.model.instance.NodeInstance;
+import org.kie.server.api.model.instance.ProcessInstance;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class StartProcessServiceIntegrationTest extends JbpmKieServerBaseIntegrationTest {
+
+    private static final KieServerConfigItem PPI_RUNTIME_STRATEGY = new KieServerConfigItem(KieServerConstants.PCFG_RUNTIME_STRATEGY, RuntimeStrategy.PER_PROCESS_INSTANCE.name(), String.class.getName());
+
+    private static ReleaseId releaseId = new ReleaseId("org.kie.server.testing", "restart-project",
+            "1.0.0.Final");
+
+    protected static final String SORT_BY_PROCESS_ID = "ProcessId";
+
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/restart-project");
+
+        kieContainer = KieServices.Factory.get().newKieContainer(releaseId);
+
+        createContainer(CONTAINER_ID_RESTART, releaseId, PPI_RUNTIME_STRATEGY);
+    }
+
+
+    @Test()
+    public void testStartProcess() {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID_RESTART, PROCESS_ID_RESTART, parameters);
+        try {
+            assertNotNull(processInstanceId);
+            assertTrue(processInstanceId.longValue() > 0);
+
+            // Process instance is running and is active.
+            ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+            assertNotNull(processInstance);
+            assertEquals(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE, processInstance.getState().intValue());
+
+            processClient.abortProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+
+            List<NodeInstance> list = this.processClient.findNodeInstancesByType(CONTAINER_ID_RESTART, processInstanceId, "ABORTED", 0, 10);
+            String[] nodeIds = list.stream().map(e -> e.getNodeId()).toArray(String[]::new);
```

Comment:
Can be replaced with method reference
```suggestion
            String[] nodeIds = list.stream().map(NodeInstance::getNodeId).toArray(String[]::new);
```

Reply:
done